### PR TITLE
add fix to sendPaperworkNewOpportunity test

### PR DESCRIPTION
**Pull request by yloson**
a dropdown button sometimes isn't available, this pr should fix it

### CODE REVIEW

File: src/test/java/pageObjects/OpportunityPage.java [hunk 1]
```diff
@@ -1,30 +1,37 @@
...
+    private By dropdownElement = By.cssSelector(".actionsContainer li.oneActionsDropDown");
     private String related = "Related";
-    private static final int WAIT_TIME_IN_SECONDS = 30;
 
     public void clickRelatedTab() {
         LOGGER.log(Level.INFO, "click Related tab in OpportunityPage");
         waitForTextWithRetries(related, 5, 5);
         waitUntilLocatorDisplayAndClick(relatedTab, 1, WAIT_TIME_IN_SECONDS);
-        explicitWaitInSeconds(WAIT_TIME_IN_SECONDS).until(ExpectedConditions.numberOfElementsToBeMoreThan(content_section,15));
+        explicitWaitInSeconds(WAIT_TIME_IN_SECONDS).until(ExpectedConditions.numberOfElementsToBeMoreThan(content_section, 15));
```
Comment (okeren-ww) [inline src/test/java/pageObjects/OpportunityPage.java]: what is this 15 value?

File: src/test/java/pageObjects/OpportunityPage.java [hunk 2]
```diff
@@ -47,9 +54,11 @@ public void accountSelector(String email) {
                 if (findElement(copyOpportunityMainContactLink).isDisplayed()) {
                     click(copyOpportunityMainContactLink);
                     waitForElementAttributeToBe(primaryMemberEmail, "value", email);
+                    waitForElementToBeClickable(accoutName, 5);
+                    click(accoutName);
                     sendKeys(accoutName, "Account-" + email);
-                  waitUntilLocatorEnableAndClick(saveButton, 1,10);
-                  explicitWaitInSeconds(45).until(ExpectedConditions.numberOfWindowsToBe(2));
+                    waitUntilLocatorEnableAndClick(saveButton, 1, 10);
```
Comment (okeren-ww) [inline src/test/java/pageObjects/OpportunityPage.java]: Also, what are these 1 & 10 values?

File: src/test/java/pageObjects/OpportunityPage.java (hunk 1 above)
Comment (yloson) [reply src/test/java/pageObjects/OpportunityPage.java]: its the number of elements to wait for which basically means that most of the page is rendered

File: src/test/java/pageObjects/OpportunityPage.java (hunk 2 above)
Comment (yloson) [reply src/test/java/pageObjects/OpportunityPage.java]: these params are minimum number of elements to be enable and timeout respectively
the **number of elements** is due to the fact that locator sometimes isn't unique (1 or more display for the same locator)

File: src/test/java/pageObjects/OpportunityPage.java [hunk 3]
```diff
@@ -1,30 +1,37 @@
...
 
     public void openActionDropDown(String value) {
-        click(actionsDropDown);
+        try {
+            clickWithEnterKey(actionsDropDown);
+            waitForElementAttributeToBe(actionsDropDown, "aria-expanded", "true");
+        } catch (NoSuchElementException | TimeoutException missingElementException) {
+            LOGGER.log(Level.INFO, "actionsDropDown was not visible, try again " + missingElementException);
+            click(actionsDropDown);
+        }
```
Comment (LilyZhouWeWork) [inline src/test/java/pageObjects/OpportunityPage.java]: correct me if I am wrong, the retry logic here is to try one more before it throws an error?
Why don't use 
retry while `element is not visible`

Comment (yloson) [reply src/test/java/pageObjects/OpportunityPage.java]: retry while element is not visible is a bit risky, when element isn't visible due to some error we'll create an endless loop in such case, the element in 95% of times is display so i am not sure a loop is required, i would go with expected condition to locate an element but as explained this is not the case.

also due to the nature of the element there are 2 different types of click here - one is simulating Enter key and the other actual click (mouse like) - this just to make sure i get the click this way or the other

File: .circleci/config.yml [hunk 4]
```diff
@@ -63,7 +63,7 @@ workflows:
   selenium-scheduled:
     triggers:
       - schedule:
-          cron: "40 * * * *"
+          cron: "55 * * * *"
```
Comment (okeren-ww) [inline .circleci/config.yml]: Maybe this should be a configuration? Seems weird that every PR you open you change this

File: src/test/java/base/BaseHelper.java [hunk 5]
```diff
@@ -56,6 +57,21 @@ public void switchFromIframe() {
         getDriver().switchTo().defaultContent();
     }
 
+    public void switchToIframeShadow(int iframeIndex) {
+        Shadow shadow = new Shadow(getDriver());
```
Comment (okeren-ww) [inline src/test/java/base/BaseHelper.java]: Making sure you actually want to make a new instance of this object every time you call this function

File: .circleci/config.yml (hunk 4 above)
Comment (yloson) [reply .circleci/config.yml]: not sure how this is done, the time changes so the run will be triggered as close as possible to the push

File: src/test/java/base/BaseHelper.java (hunk 5 above)
Comment (yloson) [reply src/test/java/base/BaseHelper.java]: good point! i will try to move this instance to be global

File: src/test/java/pageObjects/TourOutcomeFormPage.java [hunk 6]
```diff
@@ -30,10 +28,11 @@
     private By nextMonth = By.cssSelector("[title=\"Next Month\"]");
     private By datePicker = By.cssSelector(".slds-datepicker");
     private By readyToSign = By.cssSelector("[title=\"Ready to sign now\"]");
+    private By shadowLocator = By.id("winter20-astro_shadow");
```
Comment (benpony) [inline src/test/java/pageObjects/TourOutcomeFormPage.java]: "winter20-asto_shwdow" - sounds like a bad anchor as a selector 
is it a custom selector of ours or a salesforce outofthebox one ?

**Pull request merged**